src/terrama2/impl/DataRetrieverFTP.hpp

```diff
@@ -114,7 +114,7 @@ namespace terrama2
        */
       static size_t write_vector(void *ptr, size_t size, size_t nmemb, void *data);
 
-      static DataRetriever* make(DataProviderPtr dataProvider, CurlWrapper curlwrapper);
+      static DataRetriever* make(DataProviderPtr dataProvider, CurlPtr curlwrapper);
```

Comment:
Altering the constructor will affect the factory make function signature, this is not a good option.
The make signature should not be changed.